### ISSUE TITLE
A little question: why num examples is much less than the total amount of my training dataset?

### ISSUE THREAD

**Issue by Red-Scarff** [huggingface/alignment-handbook#220]
I am using this repo to SFT a model, and I notice that:

I print the total amount of my training dataset, which is 7473

`Number of raw training samples: 7473`

But during training, I find the log:

[INFO|trainer.py:2314] 2025-07-17 17:03:23,908 >> ***** Running training *****
[INFO|trainer.py:2315] 2025-07-17 17:03:23,908 >>   Num examples = 698
[INFO|trainer.py:2316] 2025-07-17 17:03:23,908 >>   Num Epochs = 3
[INFO|trainer.py:2317] 2025-07-17 17:03:23,908 >>   Instantaneous batch size per device = 2
[INFO|trainer.py:2320] 2025-07-17 17:03:23,908 >>   Total train batch size (w. parallel, distributed & accumulation) = 32
[INFO|trainer.py:2321] 2025-07-17 17:03:23,908 >>   Gradient Accumulation steps = 4
[INFO|trainer.py:2322] 2025-07-17 17:03:23,908 >>   Total optimization steps = 66
[INFO|trainer.py:2323] 2025-07-17 17:03:23,910 >>   Number of trainable parameters = 7,612,756,480

I am using a machine with 8 A100. Could anyone explain it?  I am afraid I didn't use the whole dataset but only 698 of 7473 samples to train...

**Comment by ouhenio** [comment]
Hey @Red-Scarff I faced the same situaiton. By setting `packing` to `False` in `run_sft.py` it used the entire dataset. It may be that packing is messing/filtering the dataset under the hood.

**Comment by Red-Scarff** [comment]
> Hey [@Red-Scarff](https://github.com/Red-Scarff) I faced the same situaiton. By setting `packing` to `False` in `run_sft.py` ([here](https://github.com/huggingface/alignment-handbook/blob/main/scripts/run_sft.py#L174)) it used the entire dataset. It may be that packing is messing/filtering the dataset under the hood.

thanks, I will try it.

**Comment by lewtun** [comment]
Hi @Red-Scarff and @ouhenio ! Indeed, when we set `packing=True` the N samples of the dataset are concatenated and chunked into K samples of size `max_length`. This is why you get fewer samples after packing, but it will be much more efficient to train!
